src/binaryen-c.cpp

```diff
@@ -3262,19 +3262,17 @@ void BinaryenRemoveGlobal(BinaryenModuleRef module, const char* name) {
 BinaryenEventRef BinaryenAddEvent(BinaryenModuleRef module,
                                   const char* name,
                                   uint32_t attribute,
-                                  BinaryenFunctionTypeRef type) {
+                                  BinaryenType params) {
```

Comment:
Not sure if we want to assume the result is always none here, given that Rossberg wants to use this field for some other control-flow transferring events in the future. I'm fine with fixing the code when this happens, but given that this is related to binaryen.js API, I'm not sure if we should change the API itself. WDYT?

Reply:
Do you mean we should add the `results` parameter now so we don't have to make another change in the future? SGTM.

Reply:
Are we gonna have something that corresponds to the current `BinaryenFunctionTypeRef`, such as `BinaryenSignatureRef` or something? If so, it'd be better to use it. Are we planning to divide other uses of `BinaryenFunctionTypeRef` into params and results too?

Reply:
Are we gonna have something that corresponds to the current `BinaryenFunctionTypeRef`, such as `BinaryenSignatureRef` or something? If so, it'd be better to use it. Are we planning to divide other uses of `BinaryenFunctionTypeRef` into params and results too?

Reply:
I think that for the C API it would be better to treat the params and the results as separate. That way we don't need to deal with passing a struct.